### PR TITLE
Add Barrel to LootableBlockInventory

### PR DESCRIPTION
**Pull request by TreemanKing**
Adds Barrel to the LootableBlockInventory Interface as it is currently missing.

### CODE REVIEW

File: patches/api/0040-LootTable-API.patch [hunk 1]
```diff
@@ -238,6 +238,20 @@ index 0000000000000000000000000000000000000000..fd184f13f5e8ee5cf829fff4f44696e1
...
 +    }
 +}
+diff --git a/src/main/java/org/bukkit/block/Barrel.java b/src/main/java/org/bukkit/block/Barrel.java
+index aa1bb7a1a1c94b0b029bb60026efbc7f55584dd7..b4430cb7bb8e33016a1af5fcb1cf4f671f364717 100644
+--- a/src/main/java/org/bukkit/block/Barrel.java
++++ b/src/main/java/org/bukkit/block/Barrel.java
+@@ -1,8 +1,6 @@
+ package org.bukkit.block;
+ 
+-import org.bukkit.loot.Lootable;
```
Comment (lynxplay) [inline patches/api/0040-LootTable-API.patch]: Just leave the import, even if it is unused here.

Comment (TreemanKing) [reply patches/api/0040-LootTable-API.patch]: What's the actual reason that we do this? I'm assuming it's something to do with reverting patches?

Comment (granny) [reply patches/api/0040-LootTable-API.patch]: Keeping that import adds maintenance burden to those maintaining the project. Leaving it as it was would mean the patch might fail if there's any changes to the imports in that line, even though it has no actual influence in the changes implemented by the patch.

TLDR; the less changes in a diff, the less likely for conflicts to occur, the easier it is to maintain.

File: patches/api/0040-LootTable-API.patch [hunk 2]
```diff
@@ -238,6 +238,16 @@ index 0000000000000000000000000000000000000000..fd184f13f5e8ee5cf829fff4f44696e1
...
+diff --git a/src/main/java/org/bukkit/block/Barrel.java b/src/main/java/org/bukkit/block/Barrel.java
+index aa1bb7a1a1c94b0b029bb60026efbc7f55584dd7..7ca177ff9d1c4a6b9b5d006a2f1ff1968259c92b 100644
+--- a/src/main/java/org/bukkit/block/Barrel.java
++++ b/src/main/java/org/bukkit/block/Barrel.java
+@@ -5,4 +5,4 @@ import org.bukkit.loot.Lootable;
+ /**
+  * Represents a captured state of a Barrel.
+  */
+-public interface Barrel extends Container, Lootable, Lidded { }
++public interface Barrel extends Container, com.destroystokyo.paper.loottable.LootableBlockInventory, Lidded { } //Paper
```
Comment (Lulu13022002) [inline patches/api/0040-LootTable-API.patch]: Missing a space between the // and Paper

**Review by Warriorrrr — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Review by kennytv — APPROVED** [review]
